lib/Consumer.php

```diff
@@ -68,11 +68,12 @@ public function receive(IEvent $event) {
 		$emailSetting = ($emailSetting) ? $this->userSettings->getUserSetting($event->getAffectedUser(), 'setting', 'batchtime') : false;
 
 		// Add activity to stream
-		if (!$selfAction) {
+		if ($selfAction) {
 			$activityId = $this->data->send($event);
 		}
 
-		if (!$selfAction && $notificationSetting && $activityId) {
+		//Check if the activityId is set before using the variable directly in the condition.
+		if ($selfAction && $notificationSetting && isset($activityId)  && $activityId) {
```

Comment:
```suggestion
		if (!$selfAction && $notificationSetting && $activityId) {
```

Reply:
Now I have verified with the scenario:

1. Created users `admin`, `user1`.
2. `admin` shares a file `foo.txt` to `user1`
3. `admin` comments on `foo.txt`. `admin` can see the comments and so does `user1`
4. `user1` comments on `foo.txt`. Both `admin` and `user1` can see the comments.

5. `admin` creates a file `bar.txt`
6. `admin` comments on the file`bar.txt` (self). The activity is available in the table.